fraud/proof.go

```diff
@@ -80,3 +93,39 @@ func Unmarshal(proofType ProofType, msg []byte) (Proof, error) {
 	}
 	return unmarshaler(msg)
 }
+
+// MonitorProofs enables metrics to monitor fraud proofs.
+func MonitorProofs(proofTypes ...ProofType) func(Getter) {
+	return func(store Getter) {
+		for _, proofType := range proofTypes {
+			counter, _ := meter.AsyncInt64().Counter(string(proofType),
+				instrument.WithUnit(unit.Dimensionless),
+				instrument.WithDescription("Stored fraud proof"),
+			)
```

Comment:
Counter has to be always incremented to record values, if we use Gauge here it will be possible to update metric without increasing its value. It will be easier with gauge to observe cases when no proofs or an error returned from store.